eng/Configurations.props

```diff
@@ -68,6 +70,92 @@
     <TargetsMobile Condition="'$(TargetOS)' == 'iOS' or '$(TargetOS)' == 'Android' or '$(TargetOS)' == 'tvOS' or '$(TargetOS)' == 'Browser'">true</TargetsMobile>
   </PropertyGroup>
 
+  <PropertyGroup Label="CalculateOS">
+    <_runtimeOS>$(RuntimeOS)</_runtimeOS>
+
+    <_parseDistroRid>$(__DistroRid)</_parseDistroRid>
+    <_parseDistroRid Condition="'$(_parseDistroRid)' == ''">$([System.Runtime.InteropServices.RuntimeInformation]::RuntimeIdentifier)</_parseDistroRid>
+    <_distroRidIndex>$(_parseDistroRid.LastIndexOfAny("-"))</_distroRidIndex>
+
+    <_runtimeOS Condition="'$(_runtimeOS)' == ''">$(_parseDistroRid.SubString(0, $(_distroRidIndex)))</_runtimeOS>
+
+    <!-- _runtimeOS is calculated based on the build system OS, however if building for Browser/iOS/Android we need to let
+         the build system to use browser/ios/android as the _runtimeOS for produced package RIDs. -->
+    <_runtimeOS Condition="'$(TargetsMobile)' == 'true'">$(TargetOS.ToLowerInvariant())</_runtimeOS>
+
+    <_runtimeOSVersionIndex>$(_runtimeOS.IndexOfAny(".-0123456789"))</_runtimeOSVersionIndex>
+    <_runtimeOSFamily Condition="'$(_runtimeOSVersionIndex)' != '-1'">$(_runtimeOS.SubString(0, $(_runtimeOSVersionIndex)))</_runtimeOSFamily>
+
+    <_buildingInOSX>$([MSBuild]::IsOSPlatform('OSX'))</_buildingInOSX>
+    <_portableOS>linux</_portableOS>
+    <_portableOS Condition="'$(_runtimeOS)' == 'linux-musl'">linux-musl</_portableOS>
+    <_portableOS Condition="$(_buildingInOSX)">osx</_portableOS>
+    <_portableOS Condition="'$(_runtimeOSFamily)' == 'win' or '$(TargetOS)' == 'Windows_NT'">win</_portableOS>
+    <_portableOS Condition="'$(_runtimeOSFamily)' == 'FreeBSD'">freebsd</_portableOS>
+    <_portableOS Condition="'$(_runtimeOSFamily)' == 'illumos'">illumos</_portableOS>
+    <_portableOS Condition="'$(_runtimeOSFamily)' == 'Solaris'">solaris</_portableOS>
+    <_portableOS Condition="'$(_runtimeOS)' == 'Browser'">browser</_portableOS>
+    <_portableOS Condition="'$(_runtimeOS)' == 'ios'">ios</_portableOS>
+    <_portableOS Condition="'$(_runtimeOS)' == 'tvos'">tvos</_portableOS>
+    <_portableOS Condition="'$(_runtimeOS)' == 'android'">android</_portableOS>
```

Comment:
@dagood, I was thinking that maybe it would be a good idea to move this flat list and/or `<_outputRID>` below to arcade repo in the future, in order to DRY the portable platform matrix for https://github.com/dotnet/sdk/issues/14296#issuecomment-715613987 (and other similar use-cases originating from arcade-powered repos). This will make new platforms bringup slightly smoother (without needing to update many repos individually).

In this context:
* `<PackageRID>` is the identifier used to restore packages. In cross-building scenarios (such as; building FreeBSD bits on Linux), the value of OS is `linux` to make package restore work correctly. The final output location in coreclr/libraries/tests are determined by host OS (`freebsd` in case of `FreeBSD`), which is a separate property defined elsewhere.
* `<OutputRid>` is the identifier used to determine the output directory name of installer (among other things). This is a different than how it is done for coreclr/libraries/tests, and can be simplified/removed in the future; after aligning installer with other components of runtime repo.

Reply:
I agree, centralizing this would be great! I remember that this has come up before, but it seemed unapproachable (at the time) due to the variety of ways RIDs are handled in various repos, interactions with native builds, etc. Unfortunately, I'm not able to find records of any of that or who was involved... I think the next step would be to file a dotnet/arcade issue to get input from the various people we'd want to start depending on this as an arcade feature.

Aside from how source-build needs to be able to override it, though, I don't think I'll be able to get hugely involved... I think pinging dotnet/runtime infra area owners and folks involved in SDK infra would be a reasonable start. (Not doing so here in case you want to keep this PR focused on this more local centralization for now.)

Reply:
Thanks for the tips on follow up. Yes, I would like to keep this PR focused on local centralization (internal refactoring, without changing the public contract). 

There is some follow up work required to bring about simplifications. Currently, there are multiple ways to override RID from environment variables; build scripts can influence the value of `__DistroRid` based on what was passed to `-os` flag, MSBuild property `RuntimeOS` and then `OutputRid` which is a global knob only used by installer (PR is removing its dependency from coreclr). `OutputRid` is a one-off concept, which is conditionally set to `RuntimeIdentifier`, `OfficialRidBuildRID` etc. and makes the repo-wide RID story hard to reason about. source-build is also overriding `OutputRid`, in addition to supplying `-os` as well as setting `PortableBuild` to `false`. 🤯

If we can get rid of `OutputRid` soon and agree on just one way to override the RID (via `./build.sh -os ...` which is implicitly connected to `/p:RuntimeOS=...`, when MSBuild is available on the platform), that will make this inherently-complex concept relatively easier to deal with, IMO.